src/simplify_algebra.cpp

```diff
@@ -985,20 +985,34 @@ struct find_split_reshape
         auto rsp_lens    = rsp->get_shape().lens();
         auto rsp_strides = rsp->get_shape().strides();
         rsp_strides.insert(rsp_strides.begin(), rsp_strides[0] * rsp_lens[0]);
-        auto ait = std::find(rsp_strides.begin(), rsp_strides.end(), slc_dim_size);
+
+        auto ait     = std::find(rsp_strides.begin(), rsp_strides.end(), slc_dim_size);
+        int rsp_axis = -1;
         if(ait == rsp_strides.end())
         {
             return;
         }
-        int rsp_axis = std::distance(rsp_strides.begin(), ait);
-
+        else if(ait == rsp_strides.end() - 1)
+        {
+            // edge case
+            // slice_dim == 1, in that case it could match with last stride of 1.
```

Comment:
But you are not checking that the dimension is `1`. Can you add a test with slice and the last dimension is not `1`?

Reply:
> Can you add a test with slice and the last dimension is not 1? 
Last dimension of which instruction?  
Almost all the existing tests are taking slices in the last dimension with `slice_length != 1`

Reply:
Yea lets add an `assert`.